flink-connector/flink/build.gradle.kts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+plugins {
+  `maven-publish`
+  id("java")
+  id("idea")
+}
+
+repositories {
+  mavenCentral()
+}
+
+val flinkVersion: String = libs.versions.flink.get()
+val flinkMajorVersion: String = flinkVersion.substringBeforeLast(".")
+
+
+// The Flink only support scala 2.12, and all scala api will be removed in a future version.
+// You can find more detail at the following issues:
+// https://issues.apache.org/jira/browse/FLINK-23986,
+// https://issues.apache.org/jira/browse/FLINK-20845,
+// https://issues.apache.org/jira/browse/FLINK-13414.
+val scalaVersion: String = "2.12"
+val artifactName = "${rootProject.name}-flink-${flinkMajorVersion}_$scalaVersion"
+
+dependencies {
+  implementation(project(":api"))
+  implementation(project(":catalogs:catalog-common"))
+  implementation(project(":common"))
+  implementation(project(":core"))
```

Comment:
you can add ` {
    exclude("*")
  } ` to shrink the size of lib

Reply:
I am confused why you need to depend on `core` module?

Reply:
@mchades could you create a separate PR to track this?

Reply:
LMTAL

Reply:
> @mchades could you create a separate issue to track this?

If this is just a minor fix, it is recommended to fix it in this PR.

Reply:
I'm afraid there maybe something unexpected, I will merge this PR to kick off 0.6.0